lib/3scale/backend/stats/cleaner.rb

```diff
@@ -108,6 +98,25 @@ def delete!(redis_conns, log_deleted_keys: nil)
 
           private
 
+          def handle_redis_exception(exception, redis_conn)
+            # If it's a connection error, simply log so we can continue with
+            # other shards. If it's another kind of error, it could be caused by
+            # a bug, so better re-raise.
+
+            is_connection_err = REDIS_CONN_ERRORS.any? do |conn_error|
+              exception.is_a?(conn_error)
+            end
+
+            is_timeout = exception.is_a?(Redis::CommandError) &&
+              exception.message == 'ERR Connection timed out'.freeze
```

Comment:
Can an exception be both a connection error and a timeout?

Reply:
No: https://github.com/redis/redis-rb/blob/v4.1.3/lib/redis/errors.rb